### PR TITLE
874 crd docs update workflow

### LINKED ISSUE #874: Port the CRD docs update workflow
Now that the k8ssandra-operator docs have been moved to this repo, we need to move the CRD [docs update GHA workflow](https://github.com/k8ssandra/k8ssandra/blob/main/.github/workflows/update-crd-reference.yaml) to this repo as well.
It will have to be modified to work correctly as it currently checks out the k8ssandra and k8ssandra-operator projects, which will not be necessary anymore, and it sends PR to the k8ssandra project, which we'll want to be sent to the k8ssandra-operator project.

Some of the expected changes:

- no need to checkout the  K8ssandra project: https://github.com/k8ssandra/k8ssandra/blob/main/.github/workflows/update-crd-reference.yaml#L24-L28
- work is not happening anymore in the k8ssandra directory (it's not the only place where we're referencing this folder, all the coed will need to be scanned to replace this): https://github.com/k8ssandra/k8ssandra/blob/main/.github/workflows/update-crd-reference.yaml#L40-L42
- The PR will have to be [sent](https://github.com/k8ssandra/k8ssandra/blob/main/.github/workflows/update-crd-reference.yaml#L105) to the k8ssandra-operator project (and the [existing PR check](https://github.com/k8ssandra/k8ssandra/blob/main/.github/workflows/update-crd-reference.yaml#L94) needs to target it as well)

The workflow will then have to be deleted from the [k8ssandra repo.](https://github.com/k8ssandra/k8ssandra)

### PR DESCRIPTION
**Pull request by RachanKaur**
**What this PR does**: Poting the CRD update workflow 

**Which issue(s) this PR fixes**:
Fixes #874 

**Checklist**
- [ ] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [ ] CHANGELOG.md updated (not required for documentation PRs)
- [ ] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: .github/workflows/update-crd-reference.yaml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            git checkout -f $branch
+            version=$(echo $branch | sed 's/origin\/release\///g')
+            mkdir -p ../k8ssandra/docs/content/en/reference/crd/releases/k8ssandra-operator-releases/k8ssandra-operator-crds-$version > /dev/null 2>&1
+            ~/go/bin/crdoc --resources config/crd/bases --output ../k8ssandra/docs/content/en/reference/crd/releases/k8ssandra-operator-releases/k8ssandra-operator-crds-$version/_index.md --template ../k8ssandra/docs/config/crdoc-templates/k8c.tmpl
+            sed -i "s/RELEASE_VERSION/$version/g" ../k8ssandra/docs/content/en/reference/crd/releases/k8ssandra-operator-releases/k8ssandra-operator-crds-$version/_index.md
+          done
+
+          # Generate CRD reference for K8ssandra-operator main branch (latest)
+          git checkout -f main
+          mkdir -p ../k8ssandra/docs/content/en/reference/crd/k8ssandra-operator-crds-latest/ > /dev/null 2>&1
```
Comment (adejanovski) [inline .github/workflows/update-crd-reference.yaml:54]:
```suggestion
          mkdir -p docs/content/en/reference/crd/k8ssandra-operator-crds-latest/ > /dev/null 2>&1
```

**Review by adejanovski — CHANGES_REQUESTED** [review]
There are numerous references to the k8ssandra project which haven't been modified to match the fact that this is running against the k8ssandra-operator project.
Please do a cleanup and find a way to test this before we can review the PR.
You can do this by pushing changes in your own fork, where you'll get to invoke the workflow.

File: .github/workflows/update-crd-reference.yaml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+          
+          # Fetch and cycle through K8ssandra-operator release branches to generate CRD references.
+          cd k8ssandra-operator
+          git fetch
+          k8c_releases=$(git branch -r |grep -E "release/[0-9]{1,2}.[0-9]{1,2}")
+          for branch in $k8c_releases
+          do 
+            git checkout -f $branch
+            version=$(echo $branch | sed 's/origin\/release\///g')
+            mkdir -p ../k8ssandra/docs/content/en/reference/crd/releases/k8ssandra-operator-releases/k8ssandra-operator-crds-$version > /dev/null 2>&1
```
[the remark this answers is not in the document]
Comment (RachanKaur) [reply .github/workflows/update-crd-reference.yaml:47]: do i need to change for cass-operator as well?

[the remark this answers is not in the document]
Comment (RachanKaur) [reply .github/workflows/update-crd-reference.yaml:47]: shouldn't that be /docs ?

[the remark this answers is not in the document]
Comment (RachanKaur) [reply .github/workflows/update-crd-reference.yaml]: I have updated all, testing if that works.

**Review by adejanovski — APPROVED** [review]

**Pull request merged**